std/random.d

```diff
@@ -1298,20 +1340,33 @@ random number sequences every run.
 
 Returns:
 A single unsigned integer seed value, different on each successive call
+Note:
+In general periodically 'reseeding' a PRNG does not improve its quality
+and in some cases may harm it. For an extreme example the Mersenne
+Twister has `2 ^^ 19937 - 1` distinct states but after `seed(uint)` is
+called it can only be in one of `2 ^^ 32` distinct states regardless of
+how excellent the source of entropy is.
 */
 @property uint unpredictableSeed() @trusted nothrow @nogc
 {
-    import core.thread : Thread, getpid, MonoTime;
-    static bool seeded;
-    static MinstdRand0 rand;
-    if (!seeded)
+    version (AnyARC4Random)
```

Comment:
As a conservative first option, I would suggest only using `arc4random` for `version (SecureARC4Random)`.

The consideration here is that in any case we still need to do something to improve `unpredictableSeed` for platforms that don't define `arc4random`.  Platforms that have only a legacy `arc4random` may well benefit more from sharing that longer-term solution, than from the legacy `arc4random`.  If we just switch now, the risk is that when the better solution is there, they don't get it, because everyone forgets that the legacy `arc4random` question needs revisiting.

I think it's _worth_ keeping the pressure to do better ourselves, than just outsourcing to a platform implementation that uses a deprecated standard.

Reply:
>The consideration here is that in any case we still need to do something to improve `unpredictableSeed` for platforms that don't define `arc4random`. 

These other solutions will also be platform-specific. There is nothing on the immediate horizon better than `arc4random` that will be usable on FreeBSD.

Reply:
> These other solutions will also be platform-specific. There is nothing on the immediate horizon better than `arc4random` that will be usable on FreeBSD.

OK, fair enough.

Would you be OK with the idea of breaking out the dangers-of-unwise-seeding doc into a separate patch (just to separate out the concerns)?  Otherwise, given the case you've made, I'm fine to move forward with this.

Reply:
> Would you be OK with the idea of breaking out the dangers-of-unwise-seeding doc into a separate patch (just to separate out the concerns)?

Imho there's no need for this. PRs can have doc updates too, this is already a tiny PR and its related to this PR. This has never been an _official_ criteria for a PR - it's only good practice to keep the focus of a PR small and to a single module only.

Reply:
> Imho there's no need for this. PRs can have doc updates too, this is already a tiny PR and its related to this PR.

You'll note I said separate _patch_ in the comment you are responding to.  (I did make an earlier remark about separate PR, but that was before responses that made clear that this PR should go forward essentially as-is.)